### PR TITLE
contrib: Switch from 'hub' to 'gh'

### PR DESCRIPTION
**Pull request by joestringer**
GitHub hasn't been maintaining https://github.com/github/hub for some time.
This PR switches the tree over to using the newer tool https://cli.github.com .

Tested while preparing the v1.16.0-pre.2 release.

### CODE REVIEW

**Review by michi-covalent — APPROVED** [review]
hub it and ship it

File: contrib/backporting/Dockerfile [hunk 1]
```diff
@@ -14,13 +14,14 @@ RUN apt-get install -y \
...
 
+ARG GH_VERSION=2.49.0
 RUN set -ex \
   && mkdir -p /hub \
   && cd /hub \
   && HUB_ARCH=amd64; if [ "$(uname -m)" = "aarch64" ]; then HUB_ARCH=arm64; fi \
-  && curl -L -o hub.tgz https://github.com/github/hub/releases/download/v2.14.0/hub-linux-${HUB_ARCH}-2.14.0.tgz \
-  && tar xfz hub.tgz \
-  && $(tar tfz hub.tgz | head -n1 | cut -f1 -d"/")/install \
+  && curl -L -o gh.tar.gz https://github.com/cli/cli/releases/download/v${GH_VERSION}/gh_${GH_VERSION}_linux_${HUB_ARCH}.tar.gz \
```
Comment (michi-covalent) [inline contrib/backporting/Dockerfile:22]: `linux` 😥 

edit: oh it's inside docker. never mind 👍

**Comment by michi-covalent** [comment]
... one meta feedback, might make sense to assign release-related to scripts to cilium/release-managers, and create a team for backporters so that they can own backport-related scripts? 💭

**Review by aditighag — APPROVED** [review]
Ha! Didn't know dev doctor had checks for backporter dependencies too.

**Review by aanm — COMMENTED** [review]
@joestringer any reason why we don't backport this PR?

**Comment by joestringer** [comment]
@aanm as discussed out-of-band, I'm not sure we've kept on top of backports for these scripts. It's definitely useful to have the latest version of these scripts available when making releases, so we should probably look at shifting them over to https://github.com/cilium/release and eliminate the possibility of branch-based release scripting. For now, it is possible to check out the main branch somewhere with these scripts, then run those main versions of the scripts from another directory checked out to a release branch.

**Comment by qmonnet** [comment]
Let's :+1:

**Pull request merged**